packages/compass/src/main/window-manager.ts

```diff
@@ -127,6 +127,8 @@ function showConnectWindow(
       nodeIntegration: true,
       contextIsolation: false,
       enableRemoteModule: true,
+      nodeIntegrationInWorker: true,
+      webSecurity: process.env.NODE_ENV !== 'development',
```

Comment:
Not sure what's the better way to do this. With this change, electron is able to load the worker file in local dev - avoids: `Failed to construct 'Worker': Script at 'file:///path_to_node-runtime-worker/worker-runtime.js' cannot be accessed from origin 'http://localhost:4242'.`

I tried to intercept the request but it seems the request is only happening after i add this flag.

Reply:
Hm ... @gribnoysup any thoughts? I imagine a proper fix here would involve serving the script as an asset through the webpack server, which would require changing this line here?

https://github.com/mongodb-js/mongosh/blob/0600bf36ff0b10ba63f10304df2de3d710c0dc7b/packages/node-runtime-worker-thread/src/index.ts#L69

Reply:
Yes, a proper fix would be to allow webpack to [process the worker when bundling](https://webpack.js.org/guides/web-workers/#syntax): we'd need to provide a static path to the worker runtime file in mongosh (and maybe even skip bundling in mongosh altogether, allowing compass to package the worker), although IIRC when we tried this last time, we ran into some issues, @mabaasit maybe you remember what exactly stopped us from doing this. If we can't figure out how to do this, disabling `webSecurity` here environment is probably an okay workaround for local dev environment, I would provide a very detailed comment here though explaining why we're doing it in that and why we should never do this in any other case, maybe even using a different env variable set globally for compass specifically for this purpose

Reply:
i totally don't recall the reason, but i think it was tests related somehow. as a workaround here, i added [b12622c](https://github.com/mongodb-js/compass/pull/6203/commits/b12622c6d555dcb66907ea7d926bce1a50c3afa7) as mentioned in the latter part of your comment.

Reply:
also, should i create a follow up to check if we can avoid bundling node-runtime-worker-thread on the mongosh side and then handling everything on the compass side?

Reply:
> also, should i create a follow up to check if we can avoid bundling node-runtime-worker-thread on the mongosh side and then handling everything on the compass side?

I think it might be worth it, yeah. Even if it's a short timeboxed investigation that results in us just keeping everything as-is

Reply:
added https://jira.mongodb.org/browse/COMPASS-8269